src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Extensions/IndentedStringBuilder.cs

```diff
@@ -100,16 +150,18 @@ public void AppendLine()
 		}
 
 		/// <summary>
-		/// Appends the given string, *without* appending a newline at the end.
+		/// Appends the given multi-line string with each line indented per CurentLevel, with a newline at the end.
 		/// </summary>
 		/// <param name="text">The string to append.</param>
-		/// <remarks>
-		/// Even though this method seems like it appends a newline, it doesn't. To append a
-		/// newline, call <see cref="AppendLine()"/> after this method.
-		/// </remarks>
 		public void AppendMultiLineIndented(string text)
 		{
-			_stringBuilder.Append(text.Indent(CurrentLevel));
+			// LineSplitEnumerator is a struct so there is no allocation here
+			var enumerator = new LineSplitEnumerator(text.AsSpan());
+			foreach (var line in enumerator)
+			{
+				AppendIndented(line.ToString());
```

Comment:
TODO: Make sure `.ToString()` isn't causing too much overhead. It looks like converting a span back to string isn't cheap, we need to compare this with the regex approach.

Reply:
Have you see [this](https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-6.0#system-text-stringbuilder-append(system-readonlyspan((system-char)))).

Reply:
@workgroupengineering This is not available on .NET Framework or .NET Standard 2.0

I'm looking for two alternatives, but need to do proper benchmarking.

```csharp
            unsafe
            {
                fixed (char* ptr = &MemoryMarshal.GetReference(span))
                {
                    builder.Append(ptr, span.Length);
                }
            }
```

or

```csharp
            var array = ArrayPool<char>.Shared.Rent(text.Length);
            text.CopyTo(array);
            builder.Append(array);
            ArrayPool<char>.Shared.Return(array);
```